### ISSUE TITLE
If ruby, check the version

### ISSUE THREAD

**Issue by zheileman** [ministryofjustice/repo-audit#19]

**Comment by zheileman** [comment]
I suppose it is a bit finger in the air or premature, but do we have any specific 'minimum' version in mind for Ruby repositories? i.e. >= 2.2.0

Or are we planning on going for a more elaborated solution like actually retrieving the latest stable release version and check the repo is not 1 or 2 major releases away from this, etc... I guess this could turn out to be as complicated as we can imagine and in any case will be super opinionated.

Same with issue #20 really.

**Comment by digitalronin** [comment]
These are very good points. There's also the issue of how to check what the ruby version is. It can be specified in the `Gemfile`, or there could be a `.ruby-version` file in the repo. I'm not sure if there are other ways to do it.
For now, I'd probably just enable configuring a check for something like; `~> 2.2` i.e. 'must be at least this minor release or newer.' We can think about making it cleverer later on.

**Comment by zheileman** [comment]
Yes, the problem of how to declare the ruby version might be solved by checking both places, Gemfile first (as I think is the recommended way to specify a version) and .ruby-version as a fallback.

And I agree, let's go with something simple for now 👍

**Comment by SteveMarshall** [comment]
@alan and I have been talking a little about using [Dependabot](https://dependabot.com) for this. If we're happy with using that, we could make repo-audit just check for having that configured correctly? 🤔

**Comment by digitalronin** [comment]
I definitely like the idea of plugging repo-audit into other services to do this kind of thing. A lot of these problems are much harder than they appear on the surface, and there's really no point in repeating other projects' work, so +100 for this approach.

**Comment by zheileman** [comment]
Ok, in that case we can use our existing `file_content` check, to look for a Dependabot configuration file, in a similar way to what we do for Dangerfile, if that's how it works, correct me if I'm wrong.

Definitely happy to not reinvent the wheel here, although we can still go for a simple versioning check to have something up and running, without forcing existing repos to configure Dependabot, with a prospective deprecation in favour of this tool?

For only ruby/rails (maybe python...) version checks we can do it ourselves, but obviously if we want something more elaborated, like checking all gem versions, dependencies, etc. Dependabot or similar service would help us a lot.

**Comment by digitalronin** [comment]
I think it's worthwhile to implement a basic version check, at least for ruby, and especially for rails. Running apps. based on older versions of rails is definitely something we ought to know about. 
If we have that capability in place, and a security vulnerability is discovered, we will be able to easily run a quick check for the affected version(s).

**Comment by alan** [comment]
I agree that the `Gemfile` is the preferred place to check. Is more likely that most of the projects are using `bundler`, while`.ruby-version` is used by `rbenv` so might or not be there depending on the project.

 [Dependabot](https://dependabot.com) is different, it reads the `Gemfile` and `Gemfile.lock` and opens a pull request when there is a new compatible version of a gem. Dependabot pull request [example](https://github.com/ministryofjustice/prison-visits-2/pull/930)

**Comment by digitalronin** [comment]
Yeah. The problem is that specifying a ruby version in the `Gemfile` is optional. So, we might need something that says;

There's a Gemfile, therefore it's a ruby project
   the gemfile doesn't specify ruby version, but there is a .ruby-version file, so that's the version

...or something equally baroque.

**Comment by alan** [comment]
Even though is optional in the Gemfile we could try to enforce its use?

There is also the `Dockerfile` which has information on the ruby version used in production.

**Comment by digitalronin** [comment]
Good point. We could warn "You've got a Gemfile, but it doesn't pin your ruby version."
